src/main/scala/org/broadinstitute/dsde/firecloud/core/ProfileClient.scala

```diff
@@ -208,7 +209,10 @@ class ProfileClientActor(requestContext: RequestContext) extends Actor with Fire
           RawlsGroupMemberList(userSubjectIds = Some(subjectIds))
         }}
       }
-      case Some(id) => Future.successful(RawlsGroupMemberList(userSubjectIds = Some(Seq(id))))
+      case Some((fcUser, nihUser)) =>
+        if(whitelist contains nihUser) Future.successful(RawlsGroupMemberList(userSubjectIds = Some(Seq(fcUser))))
+        else Future.failed(throw new FireCloudExceptionWithErrorReport(ErrorReport(StatusCodes.Forbidden,
```

Comment:
I'm not sure if this bubbles up in the way that we'd like but it seems to _work_ the way we want it to. Basically, if the user isn't a part of the whitelist then we don't want to do *anything* that follows. If we don't stop the operation here then we'll end up updating their thurloe profile to say that their TCGA access is "pending" when in reality we should just immediately fail and not update anything in thurloe